### DISCUSSION TITLE
Feature request: Add ability to change logical color temp range

### DISCUSSION THREAD

**Discussion by Tediore** [arendst/Tasmota#10242] [Ideas]
I have a CCT light bulb that I transplanted an ESP into that has no LED controller (it just has 2 PWMs). Tasmota's color temp range is from 2000K (500 mireds) to about 6500K (153 mireds). My custom bulb's range is from 2700K to 6500K. As a result, when I tell Tasmota to set the color temp to 2700K with Home Assistant it ends up actually being set to something around 3400K. 

I could use math to calculate a Tasmota equivalent value, but it would be nice to simply tell Tasmota the range that the bulb is capable of. Maybe commands like `CTMin` and `CTMax` where you can set the minimum and maximum mireds? In my case `CTMin` would be 153 (no change) and `CTMax` would be 370 (instead of 500).

**Comment by Tediore** [comment]
Or even just `CTRange` which could be used like `CTRange 153, 370`

**Comment by adamoberley** [comment]
This would be an awesome feature. I've noticed the difference in color temps across different Tasmotized lights when using adaptive or circadian lighting. At night when all the lights start getting warmer, some are orange while others are still close to white. If you could set the actual color temp range, it would be a lot easier to keep different lights in sync.

**Comment by s-hadinger** [comment]
Hold on, I'm about to allow more flexibility to the Virtual CT which will also allow you to do this

**Reply by vic42** [reply]
I assume some non-linearity may be observed as well. That would require something like a gamma mapping function.

**Reply by s-hadinger** [reply]
Yes, gamma correction is taken care of.

I take the RGBCW (5 virtual channels), and split into R G B W (white channel as a whole). Gamma is applied to all 4 channels, and the White power is then split between CW and WW linearly. With VirtualCT it will be the same: global Gamma applied to White, then linear distribution between required LEDS.

I take the assumption that LED brightness is linear to PWM value. Since PWM is simply on/off, I don't think there is substantial non-linearity in the LEDs themselves.

**Reply by Tediore** [reply]
Nice, thanks! Looking forward to it.

**Comment by s-hadinger** [comment]
PR submitted #10311
Just wait for it to be merged.

**Reply by Tediore** [reply]
Thanks!
